src/libostree/ostree-repo-commit.c

```diff
@@ -2496,23 +2496,16 @@ write_directory_content_to_mtree_internal (OstreeRepo                  *self,
                                            GCancellable                *cancellable,
                                            GError                     **error)
 {
-  g_autoptr(GFile) child = NULL;
-  g_autoptr(GFileInfo) modified_info = NULL;
-  g_autoptr(OstreeMutableTree) child_mtree = NULL;
-  g_autofree char *child_relpath = NULL;
-  const char *name;
-  GFileType file_type;
-  OstreeRepoCommitFilterResult filter_result;
-
   g_assert (dir_enum != NULL || dfd_iter != NULL);
 
-  name = g_file_info_get_name (child_info);
+  const char *name = g_file_info_get_name (child_info);
   g_ptr_array_add (path, (char*)name);
 
-  if (modifier != NULL)
-    child_relpath = ptrarray_path_join (path);
+  g_autofree char *child_relpath = ptrarray_path_join (path);
```

Comment:
IIRC I did this intentionally to avoid the `malloc` if we didn't have a commit modifier.  But eh, perhaps down the line we can tweak this to use a scratch `GString` buffer like we do in other places.